legos/xkcd.py

```diff
@@ -57,21 +62,31 @@ def _build_url(self, comic_id):
         if comic_id is not None:
             if comic_id == 'r' or comic_id == 'random':
                 logger.debug('Random comic requested...')
-                url = 'http://dynamic.xkcd.com/random/comic'
-            else:
+                comic_id = self._get_random_comic_id()
                 logger.debug('User requested comic by id: %s' % str(comic_id))
```

Comment:
Looks like you dropped the else statement here. The way this flowed was that if comic_id was `r` or `random`, then we'd fetch a random one. Otherwise, we'd log what ID was requested and pass that along.